docs/reference/dlm/apis/explain-data-lifecycle.asciidoc

```diff
@@ -0,0 +1,108 @@
+[[dlm-explain-lifecycle]]
+=== Explain Lifecycle API
+++++
+<titleabbrev>Explain Data Lifecycle</titleabbrev>
+++++
+
+experimental::[]
+
+Retrieves the current data lifecycle status for one or more data stream backing indices.
+
+[[dlm-explain-lifecycle-request]]
+==== {api-request-title}
+
+`GET <target>/_lifecycle/explain`
+
+[[dlm-explain-lifecycle-desc]]
+==== {api-description-title}
+
+Retrieves information about the index's current DLM lifecycle state, such as
+time since index creation, time since rollover, the lifecycle configuration
+managing the index, or any error that {es} might've encountered during the lifecycle
+execution.
+
+[[dlm-explain-lifecycle-path-params]]
+==== {api-path-parms-title}
+
+`<target>`::
+(Required, string) Comma-separated list of indices.
+
+[[dlm-explain-lifecycle-query-params]]
+==== {api-query-parms-title}
+
+`include_defaults`::
+  (Optional, Boolean) Includes default configurations related to the lifecycle of the target index.
+  Defaults to `false`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
+
+[[dlm-explain-lifecycle-example]]
+==== {api-examples-title}
+
+The following example retrieves the lifecycle state of the index `.ds-metrics-2023.03.22-000001`:
+
+[source,console]
+--------------------------------------------------
+GET .ds-metrics-2023.03.22-000001/_lifecycle/explain
+--------------------------------------------------
+// TEST[skip:we're not setting up DLM in these tests]
+
+If the index is managed by DLM `explain` will show the `managed_by_dlm` field
+set to `true` and the rest of the response will contain information about the
+lifecycle execution status for this index:
+
+[source,console-result]
+--------------------------------------------------
+{
+  "indices": {
+    ".ds-metrics-2023.03.22-000001": {
+      "index" : ".ds-metrics-2023.03.22-000001",
+      "managed_by_dlm" : true,                        <1>
+      "index_creation_date_millis" : 1679475563571,   <2>
+      "time_since_index_creation" : "843ms",          <3>
+      "rollover_date_millis" : 1679475564293,         <4>
+      "time_since_rollover" : "121ms",                <5>
+      "lifecycle" : { },                              <6>
+      "age" : "121ms"                                 <7>
```

Comment:
I think `age` is useful when it comes to retention (and potentially other future actions like downsample `after` a particular age)
One might wonder "why is this index note getting deleted" and the age abstracts having to compute the time since creation OR time since rollover (and soon - we'll have the origination date feature so it'll always be the time since origination date - irrespective of creation/rollover dates)

Reply:
Hm, I am concerned because this has caused a lot of confusion in ILM, so I am trying to see if we can express it in a less confusing way.... I do not have a good idea right now. So we can leave it as is, but I am sharing my concern in case this triggers someone else to have a better idea :)

Reply:
What about naming it closer to encapsulating what this means from the user's perspective—for example, maybe `minimum_document_age` or `minimum_possible_document_age`?

Reply:
I like your idea of "maturity", because as you said, it gets us away from the word "age", and the point about origination date (and down the road, TSDB-ness) is good also. I think either `lifecycle_maturity` or `maturity` sounds good to me.

About the "Do we need to introduce the origination date?" question, I think we will eventually need it, especially for anyone doing a re-index. (But we don't have to work that all out on this PR necessarily.)

Reply:
>  I think time_since_creation is still helpful when checking if the index was rolled over.

Oh absolutely, sorry I should've been clearer. I was proposing not displaying the `age` field (whatever it ends up being named) before `rollover`. 

Maybe `generation_lifecycle_time` is a bit more constructive (to not be confused with other TSDB related times that are tagged to each backing index)

@dakrone what do you think?

Reply:
I think `generation_time` is reasonable (I think I like it more than `generation_lifecycle_time`, but only by a little bit. I am worried that it's a little vague and will require explanation, but perhaps explanation is better than the conflation that `age` brings with it.

Reply:
Done https://github.com/elastic/elasticsearch/pull/94621/commits/789f5f65c05adb42f8f4b1c4803db8890e0c52d1

Thanks for your thoughts on this.